Kconfig.zephyr

```diff
@@ -608,6 +608,22 @@ config WARN_EXPERIMENTAL
 	  Print a warning when the Kconfig tree is parsed if any experimental
 	  features are enabled.
 
+config ENFORCE_ZEPHYR_STDINT
+	bool
+	prompt "Enforce Zephyr convention for stdint"
+	depends on !ARCH_POSIX
```

Comment:
If one goal is to avoid issues with C++ libraries, should this also be disabled when building C++ code?